### ISSUE TITLE
Produce non-git env vars even if `.git` directory is not present 

### ISSUE THREAD

**Issue by teor2345** [rustyhorde/vergen#64]
I'd like vergen to provide git env vars when there is a `.git` directory, but skip them when there is no `.git` directory.

## Actual Output

When I build vergen without a `.git` directory present, it generates an error.

If I ignore that error, it doesn't generate any env vars at all. Even the env vars that don't depend on git are skipped.

## Expected Output

Only the `Git` config depends on the `.git` directory - the other env vars can be generated without one.

So I want to be able to write a build script that works regardless of the `.git` directory:
- if there is no `.git` directory, generate all the other env vars I've asked for
- when there is a `.git` directory, generate the `Git` env vars, as well as the other env vars

## Alternative Workaround

If that's not possible, I'd like an easy way to disable the whole `Git` config. (For example, an "all off" constant for each config.)

**Comment by CraZySacX** [comment]
You can turn off the git feature.  In your `Cargo.toml` the vergen dependency line would look something like:

```vergen = { version = "5", default-features = false, features = [""build", "cargo", "rustc", "si""] }```

**Comment by teor2345** [comment]
> You can turn off the git feature. In your `Cargo.toml` the vergen dependency line would look something like:
> 
> `vergen = { version = "5", default-features = false, features = [""build", "cargo", "rustc", "si""] }`

I want the same build script and `Cargo.toml` to provide the `Git` variables if there is a `.git` directory, and the other variables if there isn't.

That way, I'm not requiring everyone who builds the code to have a `.git` directory, but I can still use it for debug info when it is present.

(I've edited the ticket description to make that clearer.)

**Comment by CraZySacX** [comment]
Ah, I see.  This can be done in one build script today.

```
let config = if fs::metadata(".git").is_ok() {
   // Setup your config without git enabled
} else {
    // Setup your config with git enabled
    Config::default()
};

vergen(config)?
```

When writing this up I did notice that there is no good way to disable a feature programmatically, so I will be adding that.

**Comment by teor2345** [comment]
Thanks for this change, here's what we ended up doing:

```rust
    match vergen(config) {
        Ok(_) => {}
        Err(e) => {
            eprintln!(
                "git error in vergen build script: skipping git env vars: {:?}",
                e
            );
            *config.git_mut().enabled_mut() = false;
            vergen(config).expect("non-git vergen should succeed");
        }
```
https://github.com/ZcashFoundation/zebra/pull/2065/commits/c7d4bac594ea51e4a08be2fe3a194448d4162dde#diff-068676a96051f451abd4ca1d1b502d1e38a01c91a614374c0b54c6db75098079R35-R45

We can't check for `.git` because we're in a workspace / monorepo (#30, #21). And we don't want to try to replicate `vergen`'s checks, because that's fragile. (In the specific case we tested, it was actually `peel_to_commit()` that failed when there was no `.git`.)

So far, running `vergen()` twice with different configs doesn't seem to have caused any issues.
